Make beatmap importing async.

peppy opened the pull request.

peppy committed:
Make beatmap importing async.

Tom94 committed:
Merge branch 'master' into async-imports

Tom94 commented:
I don't think this is ready yet: having multiple beatmaps imported at once messes with the SQLite transaction in BeatmapDatabase.cs line 172

peppy committed:
Lock the BeatmapDatabase's connection during imports.

This should avoid any potential issues with intertwined transactions on the same connection while still allowing higher throughput when importing.

peppy committed:
Merge branch 'master' into async-imports

Tom94 commented on osu.Game/Database/BeatmapDatabase.cs:
```diff
@@ -181,15 +181,18 @@ public void Import(string path)
-            connection.BeginTransaction();
-
-            foreach (var s in beatmapSets)
+            lock (connection)
```
The connection is also used in other places. I don't think this is thread safe at all. Maybe the content of this method should be scheduled to the main thread instead of locking?
